indra/newview/lluilistener.cpp

```diff
@@ -302,9 +311,19 @@ void LLUIListener::addMenuItem(const LLSD&event) const
     item_params.on_click = item_func;
     if(LLMenuGL* parent_menu = get_parent_menu(response, event))
     {
-        if(!parent_menu->append(LLUICtrlFactory::create<LLMenuItemCallGL>(item_params)))
+        auto item = LLUICtrlFactory::create<LLMenuItemCallGL>(item_params);
+        // Clamp pos to getItemCount(), meaning append. If pos exceeds that,
+        // insert() will silently ignore the request.
+        auto pos = get_pos(event, parent_menu->getItemCount());
```

Comment:
Correct me if I'm wrong, but after clamping to getItemCount() here we'll still to insert() and it will fail.